lib/auth/auth_with_roles.go

```diff
@@ -4833,6 +4833,27 @@ func (a *ServerWithRoles) UpsertLock(ctx context.Context, lock types.Lock) error
 	if err := a.action(apidefaults.Namespace, types.KindLock, types.VerbCreate, types.VerbUpdate); err != nil {
 		return trace.Wrap(err)
 	}
+
+	if lock.CreatedBy() == "" {
+		var createdBy string
+		hasAdmin := a.hasBuiltinRole("Admin")
+		if hasAdmin {
+			createdBy = "admin"
+		} else {
+			user, err := a.GetCurrentUser(ctx)
```

Comment:
User should be available via `a.context.User` without making an additional API call.

Reply:
Thanks, updated